version: '3.7'

services:
  postgres:
    image: postgres:13.4-alpine
    volumes:
      - postgres_data:/var/lib/postgresql/data/
    env_file:
      - config/.env.auth

  nginx:
    image: nginx:1.20
    ports:
      - "80:80"
    volumes:
      - ./src/auth/nginx/default.conf:/etc/nginx/conf.d/default.conf
      - ./src/auth/nginx/nginx.conf:/etc/nginx/nginx.conf
    depends_on:
      - auth_api

  auth_api:
    container_name: auth_api
    image: ya_practicum/auth_api
    build:
      context: ./
      dockerfile: docker/auth/Dockerfile
    restart: always
    depends_on:
      - postgres
      - redis
    env_file:
      - config/.env.auth
    entrypoint: >
      sh -c " python3 /services/wait_for_postgres.py
      && alembic upgrade head
      && gunicorn wsgi_app:app -b 0.0.0.0:8000 -w 2 --log-level debug --timeout 0"
    networks:
      - default
      - movies-network

  rmq_subscriber:
    image: ya_practicum/auth_api
    env_file:
      - config/.env.auth
    command: python3 /db/subscriber.py
    networks:
      - default
      - movies-network

  redis:
    container_name: redis_container
    image: redis:6.2.6

networks:
  movies-network:
    external: true

volumes:
  postgres_data: